plugins/modules/cloud/lxd/lxd_profile.py

```diff
@@ -307,10 +328,96 @@ def _needs_to_apply_profile_configs(self):
             self._needs_to_change_profile_config('devices')
         )
 
-    def _apply_profile_configs(self):
-        config = self.old_profile_json.copy()
+    def _merge_dicts(self, source, destination):
+        """Merge Dictionary
+
+        Get a list of filehandle numbers from logger to be handed to
+        DaemonContext.files_preserve
+
+        Args:
+            dict(source): source dict
+            dict(destination): destination dict
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            dict(destination): merged dict"""
+        for key, value in source.items():
+            if isinstance(value, dict):
+                # get node or create one
+                node = destination.setdefault(key, {})
+                self._merge_dicts(value, node)
+            else:
+                destination[key] = value
+        return destination
+
+    def _merge_config(self, config):
+        """ merge profile
+
+        Merge Configuration of the present profile and the new desired configitems
+
+        Args:
+            dict(config): Dict with the old config in 'metadata' and new config in 'config'
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            dict(config): new config"""
+        # merge or copy the sections from the existing profile to 'config'
+        for item in ['config', 'description', 'devices', 'name', 'used_by']:
+            if item in config:
+                config[item] = self._merge_dicts(config['metadata'][item], config[item])
+            else:
+                config[item] = config['metadata'][item]
+        # merge or copy the sections from the ansible-task to 'config'
+        return (self._merge_dicts(self.config, config))
+
+    def _generate_new_config(self, config):
+        """ rebuild profile
+
+        Rebuild the Profile by the configuration provided in the play.
+        Existing configurations are discarded.
+
+        This ist the default behavior.
+
+        Args:
+            dict(config): Dict with the old config in 'metadata' and new config in 'config'
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            dict(config): new config"""
         for k, v in self.config.items():
             config[k] = v
+        return(config)
```

Comment:
```suggestion
        return config
```

Reply:
Thanks, I have included the suggestion in the code.